src/memfd.rs

```diff
@@ -192,21 +198,44 @@ impl Memfd {
     /// Return the current sealing bitflags.
     fn file_get_seals(fp: &fs::File) -> Result<u64, crate::Error> {
         let fd = fp.as_raw_fd();
-        // UNSAFE(lucab): required syscall.
-        let r = unsafe { libc::syscall(libc::SYS_fcntl, fd, libc::F_GET_SEALS) };
+        // SAFETY: The syscall called has no soundness implications (i.e. does not mess with
+        // process memory in weird ways, checks its arguments for correctness, etc.). Furthermore
+        // due to invariants of `Memfd` this syscall is provided a valid file descriptor.
+        let r = unsafe {
+            libc::syscall(libc::SYS_fcntl, fd, libc::F_GET_SEALS)
+        };
         if r < 0 {
             return Err(crate::Error::GetSeals(std::io::Error::last_os_error()));
         };
-
         Ok(r as u64)
     }
+}
+
+impl AsRawFd for Memfd {
```

Comment:
@sdroege do you really really really strictly need this? (really :smiley:)
I've had a few not-so-pleasant experiences in the past which involved this trait, because it basically copies a FD without really doing a `dup()`. As a result, you can have two objects pretending they can own (and lifecycle/drop) the same FD.

I'd very much rather prefer sticking to a combination of IntoRawFd+TryFrom<RawFd> only.

Reply:
I don't need it necessarily, but in my experience it can be useful to get the fd out of things without destroying them to be able to call `fnctl` or similar on them for custom things :)

Reply:
I surely see the usefulness of that, but this kind of scenario is exactly what concerns me: through this you can (mistakenly) mutate/duplicate/invalidate the FD without taking a mutable borrow or full ownership of it. My feedback is that this trait is basically an anti-pattern which bypasses normal Rust rules.

Reply:
Note that this functionality is already available as a public API by going through `as_file`, so I don’t particularly care too much about _this_ or the `IntoRawFd` as much – they have equivalents that are as _good_, maybe just require more typing.

If it’d allow this (including `FromRawFd`) land sooner, I'm open to just removing these APIs.